core/domain/subtopic_page_services.py

```diff
@@ -0,0 +1,195 @@
+            linked.
+        include_deleted: bool. Whether to include the subtopic pages that
+            were not strictly deleted.
+
+    Returns:
+        list(SubtopicPage). The list of subtopic pages linked to the given topic
+            id.
+    """
+    subtopic_page_models = topic_models.SubtopicPageModel.query().filter(
+        (topic_models.SubtopicPageModel.topic_id == topic_id) and
+        (topic_models.SubtopicPageModel.deleted == include_deleted))
+    subtopic_pages = [
+        get_subtopic_page_from_model(subtopic_page_model)
+        for subtopic_page_model in subtopic_page_models
+    ]
+    return subtopic_pages
+
+
+def save_subtopic_page(
+        committer_id, subtopic_page, commit_message, change_list):
+    """Validates a subtopic page and commits it to persistent storage. If
+    successful, increments the version number of the incoming subtopic page
+    domain object by 1.
+
+    Args:
+        committer_id: str. ID of the given committer.
+        subtopic_page: SubtopicPage. The subtopic page domain object to be
+            saved.
+        commit_message: str. The commit message.
+        change_list: list(SubtopicPageChange). List of changes applied to a
+            subtopic page.
+
+    Raises:
+        Exception: Received invalid change list.
+        Exception: The subtopic page model and the incoming subtopic page domain
+            object have different version numbers.
+    """
+    if not change_list:
+        raise Exception(
+            'Unexpected error: received an invalid change list when trying to '
+            'save topic %s: %s' % (subtopic_page.id, change_list))
+    subtopic_page.validate()
+
+    subtopic_page_model = topic_models.SubtopicPageModel.get(
+        subtopic_page.id, strict=False)
+    if subtopic_page_model is None:
+        subtopic_page_model = topic_models.SubtopicPageModel(
+            id=subtopic_page.id)
+    else:
+        if subtopic_page.version > subtopic_page_model.version:
+            raise Exception(
+                'Unexpected error: trying to update version %s of topic '
+                'from version %s. Please reload the page and try again.'
+                % (subtopic_page_model.version, subtopic_page.version))
+        elif subtopic_page.version < subtopic_page_model.version:
+            raise Exception(
+                'Trying to update version %s of topic from version %s, '
+                'which is too old. Please reload the page and try again.'
+                % (subtopic_page_model.version, subtopic_page.version))
+
+    subtopic_page_model.topic_id = subtopic_page.topic_id
+    subtopic_page_model.html_data = subtopic_page.html_data
+    subtopic_page_model.language_code = subtopic_page.language_code
+    change_dicts = [change.to_dict() for change in change_list]
+    subtopic_page_model.commit(committer_id, commit_message, change_dicts)
+    subtopic_page.version += 1
+
+
+def update_html_data(
```

Comment:
It's odd to pass all subtopic pages into this function if you're only going to use one. I think you should just do subtopic_page.update_html_data() in whatever is calling this function.

Reply:
Done, removed the function. This was only called from apply_change_list. Moved some of the logic up to it.